src/Mesh/Topography.jl

```diff
@@ -0,0 +1,173 @@
+  
+    
+    returns nlon, nlat, lonmin, lonmax, latmin, latmax, dlon, dlat
+
+ 
+"""
+function ReadExternalHeader(header_file_in)
+    
+    @info @sprintf """ Topography header file %s ... DONE""" header_file_in
+    
+    ftopo_header = open(header_file_in)
+    @info @sprintf """ Grids.jl: Opening topography header file ... DONE"""
+    
+    topo_header = readdlm(ftopo_header)
+    @info @sprintf """ Grids.jl: Reading topography header file ... DONE""" 
+
+    #Check that the file is not empty
+    (nrows, _) = size(topo_header)    
+
+    close(ftopo_header)           
+    @info @sprintf """ Grids.jl: Closing topography header file ... DONE"""
+
+    
+    if nrows == 0
+        error(" ERROR Grids.jl --> READTOPOtxt_header: grid header file is empty!")
+        return
+    end
+    
+    nlon      =   Int64(topo_header[1,2])
+    nlat      =   Int64(topo_header[2,2])
+    lonmin    = Float64(topo_header[3,2])
+    latmin    = Float64(topo_header[4,2])
+    deltacell = Float64(topo_header[5,2])
+    lonmax    = lonmin + nlon*deltacell
+    latmax    = latmin + nlat*deltacell
+    
+    dlon  = lonmax - lonmin
+    dlat  = latmax - latmin
+
+    return (nlon, nlat, lonmin, lonmax, latmin, latmax, dlon, dlat)
+    
+end
+# }}}
+
+
+# {{{ READTOPOtxt_header
+"""
+    ReadExternalTxtCoordinates(body_file_in, TopoBathy_flg, nlon, nlat)
+
+    Reads the topography from a NOAA text file of shape [1:nnodes][3]
+    where the first and second column are the ordered lat-lon coordinates
+    and the third column is the height of topography at that specific
+    coordinate point.
+                     
+    1) XYZ files from NOAA
+                       
+     READTOPOtxt_header() reads the parameters from the header file (*.hdr)
+     READTOPOtxt_file()   reads the actual file of coordinates      (*.xyz)
+                         
+     2) DEM files
+                      
+     READTOPO_DEM()       reads a DEM file from NOAA page (file extension: *.asc)
+                     
+    returns xTopo1d, yTopo1d, zTopo2d
+
+ """
+function ReadExternalTxtCoordinates(body_file_in, TopoBathy_flg, nlon, nlat)
+
+    @info @sprintf """ Topography file: %s""" body_file_in
+
+    DFloat  = Float64 #NOTICE: I recommend the grid is always build in double precision even if the code is run in single 
+     
+    ftopo_body = open(body_file_in)
+    @info @sprintf """ Grids.jl: Opening topography file ... DONE"""
+   
+    topo_body = readdlm(ftopo_body)
+    (npoin_linear_grid, _) = size(topo_body)
+    if npoin_linear_grid == 0
+        error(" ERROR Grids.jl --> READTOPOtxt_xyz: grid file is empty!")
+        return
+    end
+    #
+    # Create array
+    #
+    nnodes_lon, nnodes_lat, nnodes = nlon, nlat, nlon*nlat #Linear grid
+    zTopo2d = Array{DFloat, 2}(undef, nnodes_lon, nnodes_lat)
+    
+    xTopo1d = Array{DFloat, 1}(undef, nnodes_lon)
+    yTopo1d = Array{DFloat, 1}(undef, nnodes_lat)
+    
+    k = 0
+    for j = 1:nnodes_lat
```

Comment:
This for loop code be replaced with just a reshape
```julia
zTopo2d = reshape(topo_body[:,3], nnodes_lon, nnodes_lat);
```

Reply:
will change this.